ui/sections/storage/inspectors/volume-dataset.reel/volume-dataset.js

```diff
@@ -80,6 +80,23 @@ exports.VolumeDataset = AbstractInspector.specialize(/** @lends VolumeDataset# *
             }
             var storageService = this.application.storageService;
             storageService.initializeDatasetProperties(this.object);
+
+            this.addPathChangeListener('object.type', this, '_handleExtraValidation');
```

Comment:
you should add listeners only if first parameter of `enterDocument` (named `isFirstTime` by convention) is true and you don't have to remove them in `exitDocument`